tests/auth_basic_test.py

```diff
@@ -64,41 +64,76 @@ def test_signup_route(client, capture_templates, setup_logger):
+        ("test@example.com", "wrongpassword", "wrongpassword"),
+        # SQL Injections
+        ("test@example.com' OR '1'='1' --", "TestP@ssword!", "TestP@ssword!"),
+        (
+            "test@example.com",
+            "wrongpassword' UNION SELECT 1, username, password FROM users --",
+            "wrongpassword' UNION SELECT 1, username, password FROM users --",
+        ),
+    ],
+)
+def test_signup_post_invalid_email(
+    client, capture_templates, setup_logger, email, password, password_confirm
+):
     """Tests the signup with an invalid email format.
 
-    Verifies that the system
-    correctly identifies the email as invalid and returns to the signup page.
+    Verifies that the system correctly identifies the email as invalid and
+    returns to the signup page.
 
     Args:
         client: The test client used for making requests.
         capture_templates: Context manager to capture templates rendered.
         setup_logger: Setup logger.
+        email: test email to be used in the signup form.
+        password: test password to be used along with the email in the signup
+                form.
+        password_confirm: test confirm password to be used along with the email
+                        in the signup form.
     """
     logger = setup_logger("test_signup_post_invalid_email")
-    try:
-        response = client.post(
-            "/signup",
-            data={
-                "email": "bad_email",  # invalid email format
-                "password": "TestP@ssword!",
-            },
-            follow_redirects=True,
-        )
-        assert response.status_code == 200
-        assert b"Please enter a valid email address" in response.data
-        assert len(capture_templates) == 1
-        final_template_rendered = len(capture_templates) - 1
-        assert (
-            capture_templates[final_template_rendered][0].name == "signup.html"
-        ), "Wrong template used"
-        logger.info("Sign up failed successfully with invalid email.")
-    except AssertionError as e:
-        logger.error(f"Test failed: {str(e)}")
-        raise
-
-
-def test_signup_post_invalid_password(client, capture_templates, setup_logger):
+    response = client.post(
+        "/signup",
+        data={
+            "email": email,
+            "password": password,
+            "password_confirm": password_confirm,
+        },
+        follow_redirects=True,
+    )
+    assert response.status_code == HTTPStatus.OK
+    assert b"Please enter a valid" in response.data
+
+    assert len(capture_templates) == 1
+    final_template_rendered = len(capture_templates) - 1
+    assert (
+        capture_templates[final_template_rendered][0].name == "signup.html"
+    ), "Wrong template used"
+    logger.info("Sign up failed successfully with invalid email.")
+
+
+@pytest.mark.parametrize(
+    "email, password, password_confirm",
+    [
+        ("bad_email", "TestP@ssword!", "TestP@ssword!"),
```

Comment:
🔬 Since you're using these multiple times, I think they're worth putting into a constant.